src/basic/user-util.c

```diff
@@ -450,6 +451,59 @@ int in_group(const char *name) {
         return in_gid(gid);
 }
 
+size_t merge_gid_lists(
+        const gid_t *list1,
+        size_t size1,
+        const gid_t *list2,
+        size_t size2,
+        gid_t *result) {
+
+        size_t nresult = 0;
+
+        /* Copying in the same order - kernel will sort it anyway in setgroups */
+        if (result) {
+                memcpy_safe(result, list1, sizeof(*list1)*size1);
+                result += size1;
+        }
+        nresult = size1;
+
+        /* We don't want any duplicates passed to setgroups, as they'd be stored in the kernel. */
+        for (size_t l2=0; l2<size2; ++l2) {
+                bool found = false;
+                for (size_t l1=0; l1 < size1; l1++) {
+                        if (list1[l1] == list2[l2]) {
+                                found = true;
+                                break;
+                        }
+                }
+                if (!found) {
+                        if (result) {
+                                *result++ = list2[l2];
+                        }
+                        ++nresult;
+                }
+        }
+        return nresult;
+}
+
+int get_current_gids(gid_t** gids) {
+        gid_t *ids = NULL;
+        size_t ngids;
```

Comment:
`int` is used as the return type in `getgroups`. Let's keep things consistent. Don't use `size_t`.